### PR TITLE
Marginally better display of refusal reason in trade UI

### PR DESCRIPTION
**Pull request by andrei8l**
#### Summary
None

#### Purpose of change
There's no indication why some traders refuse to sell or buy items

#### Describe the solution
1. Switch `npc::wants_to_sell()` and `npc::wants_to_buy()` to `ret_val<bool>`
2. Add a refusal message to `shopkeeper_item_group` and `shopkeeper_blacklist`

TODO: 
- [x] extract refusal for translation

#### Describe alternatives you've considered
Ideally, the EOC conditions should be auto-stringified like in Paradox games but I don't think we're quite there yet.

#### Testing
<details>
<summary>before</summary>

Screenshot from 2022-06-20 23-32-46
Ok, but why?
</details>

<details>
<summary>after</summary>

Screenshot from 2022-06-20 23-31-51
I didn't want it anyway
</details>

#### Additional context
N/A

### CODE REVIEW

**Review by bombasticSlacks — APPROVED** [review]
The CPP for this looks good to me, but I'm not comfortable enough with the parser to merge it without more input.

File: lang/string_extractor/parsers/shop_blacklist.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from ..write_text import write_text
+
+
+def parse_shopkeeper_blacklist(json, origin):
+    for entry in json["entries"]:
+        if "message" in entry:
+            comment = entry.get("//", None)
+            write_text(entry["message"], origin,
+                       comment=["Reason for a blacklist entry", comment])
```
Comment (BrettDong) [inline lang/string_extractor/parsers/shop_blacklist.py:9]: It'd be better mention shopkeeper here in the comment.

**Review by BrettDong — APPROVED** [review]
Translation part looks good to me.

**Comment by andrei8l** [comment]
Thanks for merging.

**Pull request merged**